File: baekjoon/10828_스택.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;
int main() {
	int N;
	string what;
	int x = 0;
	stack<int> s;
	cin >> N;
	while (N--) {
		cin >> what;
		if (what == "push") {
			cin >> x;
			s.push(x);
		}
		else if (what == "pop") {
			if (!s.empty()){
				cout << s.top() << endl;
				s.pop();
			}
			else{
				cout << -1 << endl;
			}
		}
		else if (what == "size") {
			cout << s.size() << endl;
		}
		else if (what == "empty") {
			if (s.empty()) cout << 1 << endl;
			else cout << 0 << endl;
		}
		else if (what == "top") {
			if (s.empty()) cout << -1 << endl;
			else cout << s.top() << endl;
		}
	}
}

File: baekjoon/10828_스택.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		Stack<Integer> stack = new Stack<>();
		int N = in.nextInt();
		while (N-- > 0) {
			String op = in.next();
			if (op.equals("push")) {
				int temp = in.nextInt();
				stack.push(temp);
			} else if (op.equals("pop")) {
				if (stack.isEmpty())
					System.out.println(-1);
				else
					System.out.println(stack.pop());
			} else if (op.equals("size")) {
				System.out.println(stack.size());
			} else if (op.equals("empty")) {
				if (stack.isEmpty())
					System.out.println(1);
				else
					System.out.println(0);
			} else if (op.equals("top")) {
				if (stack.isEmpty())
					System.out.println(-1);
				else
					System.out.println(stack.peek());
			}
		}
	}
}